client/src/components/mixins/dirtyGuard.js

```diff
@@ -1,36 +1,36 @@
-import DiscardChangesDialog from "../dialogs/DiscardChangesDialog.vue";
+import DiscardChangesDialog from "../dialogs/DiscardChangesDialog.vue"
 
 export default {
   methods: {
-    beforeUnload: function(e) {
+    beforeUnload: function (e) {
       if (this.dirty) {
-        e.preventDefault();
-        e.returnValue = "";
+        e.preventDefault()
+        e.returnValue = ""
       }
     },
 
-    dirtyDialog: function() {
+    dirtyDialog: function () {
       return this.$q.dialog({
-        component: DiscardChangesDialog
-      });
-    }
+        component: DiscardChangesDialog,
+      })
+    },
   },
-  beforeRouteLeave: function(to, from, next) {
+  beforeRouteLeave: function (to, from, next) {
     if (!this.dirty) {
-      return next();
+      return next()
     }
     this.dirtyDialog()
-      .onOk(function() {
-        next();
+      .onOk(function () {
+        next()
       })
-      .onCancel(function() {
-        next(false);
-      });
+      .onCancel(function () {
+        next(false)
+      })
+  },
+  mounted: function () {
+    window.addEventListener("beforeUnload", this.beforeUnload)
   },
-  mounted: function() {
-    window.addEventListener("beforeUnload", this.beforeUnload);
+  beforeDestroy: function () {
+    window.removeEventListener("beforeUnload", this.beforeUnload)
```

Comment:
I generally find this file in difficult to understand and I'd love to review it with you again at some point in the future.